### PR TITLE
Add discussion ADR for Adoping Nix Flakes

### PR DESCRIPTION
**Pull request by borancar**

### CODE REVIEW

File: docs/adrs/adr-004-nix-flakes.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# ADR 4: Adopting Nix Flakes
+
+## Context
+
+We're currently using Nix flakes, which has lead to some friction with adoption - users new to Nix see an added complexity. There has been a call to simplify with refactoring Flakes into Nix expressions.
+
+### tl;dr Points
+
+Points pro de-flaking:
+- Easier adoption, less friction, the ability to "dip feet" vs "dive-in" with impure shells.
```
Comment (AleksanderGondek) [inline docs/adrs/adr-004-nix-flakes.md:10]: Would perhaps adding "--impure" switch in our setup, help with adoption?

**Comment by AleksanderGondek** [comment]
I have some further questions, in regards to what have recently caused some friction. 
I hope it does not make me seem like I am trying to be difficult - quite the contrary, I am trying to build an understanding of what the issues are, so that I can start thinking about a solution.

From what I understand:

- `--pure` is annoying and there should be a way around it
- Flake mechanism adds additional learning curve in terms of updating the point of reference for nix definitions
- ?? Is there something more ??

**Comment by borancar** [comment]
Just those 2 sadly. No, you're not being difficult, you're more preaching to the choir than being difficult :)

File: docs/adrs/adr-004-nix-flakes.md (hunk 1 above)
Comment (borancar) [reply docs/adrs/adr-004-nix-flakes.md:10]: Yup, awesome work on https://github.com/Qarik-Group/disruptor/pull/23.

**Comment by AleksanderGondek** [comment]
Last call for comments / insights

**Pull request merged**